### PR TITLE
Oscar/setext

### PR DESCRIPTION
**Pull request by odiazib**
This PR contains the port to c++ of [`setext`](https://github.com/eagles-project/e3sm_mam4_refactor/blob/9530f6a313cf7689f542880021b5d27a6f06f3ce/components/eam/src/chemistry/mozart/mo_setext.F90#L71C5-L71C5) and [`extfrc_set`](https://github.com/eagles-project/e3sm_mam4_refactor/blob/9530f6a313cf7689f542880021b5d27a6f06f3ce/components/eam/src/chemistry/mozart/mo_extfrc.F90#L297), where `setext` invokes `extfrc_set`. 

Note: Input variables (or fields) in the subroutine [`outfld`](https://github.com/eagles-project/e3sm_mam4_refactor/blob/9530f6a313cf7689f542880021b5d27a6f06f3ce/components/eam/src/chemistry/mozart/mo_extfrc.F90#L355) (diagnostic output variables) were not included in this port. As a result, `setext` and `extfrc_set` are equivalent, so I recommend to use `extfrc_set` instead of `setext` in mam4xx. In addition, the input `zint` was removed, because it is only required to compute diagnostic output variables in `setext` and `extfrc_set`. 

This port passed validation tests with a relative tolerances of 9e-10. Validation files were generated by @tangshuaiqi.

### CODE REVIEW

File: src/mam4xx/mo_setext.hpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+namespace mam4 {
+
+namespace mo_setext {
+using View1D = DeviceType::view_1d<Real>;
+using View2D = DeviceType::view_2d<Real>;
+using View3D = DeviceType::view_3d<Real>;
+constexpr int pver = mam4::nlev;
+constexpr int extfrc_cnt = 9;
+constexpr int extcnt = 9; //, & ! number of species with external forcing
```
Comment (mjs271) [inline src/mam4xx/mo_setext.hpp:17]: Probably trivial or not worth doing now, but it appears `extcnt` is never used in a meaningful way.

**Review by mjs271 — APPROVED** [review]
Looks solid. Only one non-critical comment.

**Comment by jeff-cohere** [comment]
Thanks @odiazib ! Is there a corresponding PR for the `mam_x_validation` repo, or have you already merged the updated input files in the `main` branch there?

File: src/mam4xx/mo_setext.hpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  --------------------------------------------------------*/
+  // param[in] forcings(extcnt) array with a list of Forcing object.
+  // @param[out] frcing(ncol,pver,extcnt)   insitu forcings [molec/cm^3/s]
+  // Note: we do not need zint to compute frcing
+  // const ColumnView &zint
+
+  constexpr Real zero = 0.0;
+
+  // frcing(:,:,:) = zero;
+  if (extfrc_cnt < 1 || extcnt < 1) {
```
Comment (jeff-cohere) [inline src/mam4xx/mo_setext.hpp:40]: @mjs271: it looks like `extcnt` is used here.

**Review by jeff-cohere — APPROVED** [review]

**Comment by odiazib** [comment]
> Thanks @odiazib ! Is there a corresponding PR for the `mam_x_validation` repo, or have you already merged the updated input files in the `main` branch there?

I created [PR #52](https://github.com/eagles-project/mam_x_validation/pull/52#issue-1991557212 in mam_x_validation)

**Comment by jeff-cohere** [comment]
> > Thanks @odiazib ! Is there a corresponding PR for the `mam_x_validation` repo, or have you already merged the updated input files in the `main` branch there?
> 
> I created [PR #52](https://github.com/eagles-project/mam_x_validation/pull/52#issue-1991557212) in mam_x_validation

Approved, thanks!

**Pull request merged**